File: pdf-ai-summarizer/backend/app/schemas/summary.py
from pydantic import BaseModel

class SummaryRequest(BaseModel):
    document_id: str
    document_name: str
    summary_type: str
    language: str
    max_length: int
    audience: str
    focus: str
    tone: str
    custom_note: str = ""
    extraction_mode: str = "text"

class SummaryResponse(BaseModel):
    document_id: str
    summary: str

class PdfExportRequest(BaseModel):
    document_id: str

File: pdf-ai-summarizer/frontend/types/summary.ts
export type SummaryRequest = {
  document_id: string;
  document_name: string;
  summary_type: string;
  language: string;
  max_length: number;
  audience: string;
  focus: string;
  tone: string;
  custom_note: string;
  extraction_mode: string;
};

export type SummaryResponse = {
  document_id: string;
  summary: string;
};
